# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/canvas/WebGLProgram.h
# Commit: 7e2a2b1b416f
# Full Hash: 7e2a2b1b416f1a64c6548aab29e92acc9554ae7d
# Author: Jeff Gilbert <jgilbert@mozilla.com>
# Date: 2020-01-09 03:40:18
# Regressor Bug: 1477756
# File Overlap Count: 1
# Description:
#   Bug 1477756 - Fix all webgl regression tests according to CI. r=handyman
#   
#   (This is a combination of 31 commits)
#   
#   * Fix Linux compilation.
# ==============================================================================

diff -r ccfa767dba64 -r 7e2a2b1b416f dom/canvas/WebGLProgram.h
--- a/dom/canvas/WebGLProgram.h	Wed Jan 08 22:19:16 2020 +0000
+++ b/dom/canvas/WebGLProgram.h	Wed Jan 08 22:19:23 2020 +0000
@@ -60,6 +60,7 @@
 
 struct ActiveUniformValidationInfo final {
   const ActiveUniformInfo& info;
+  bool isArray = false;
   uint8_t channelsPerElem = 0;
   decltype(&UniformAs1fv) pfn = nullptr;
 
@@ -99,7 +100,7 @@
 
   mutable std::vector<size_t> componentsPerTFVert;
 
-  bool attrib0Active;
+  bool attrib0Active = false;
 
   // -
 
@@ -146,7 +147,7 @@
 
   void LinkProgram();
   bool UseProgram() const;
-  void ValidateProgram() const;
+  bool ValidateProgram() const;
 
   ////////////////
 